viewer/packages/data-providers/src/image-360-data-providers/descriptor-providers/events/Cdf360BatchEventCollectionLoader.ts

```diff
@@ -0,0 +1,312 @@
+        .filter(fileInfoSets => fileInfoSets.length === 6);
+    }
+    return [fileInfos];
+  }
+
+  private parseEventMetadata(
+    eventMetadata: Event360Metadata,
+    preMultipliedRotation: boolean
+  ): Image360RevisionDescriptor<ClassicDataSourceType> {
+    return {
+      collectionId: eventMetadata.site_id,
+      collectionLabel: eventMetadata.site_name ?? undefined,
+      id: eventMetadata.station_id,
+      label: eventMetadata.station_name ?? undefined,
+      transform: this.parseTransform(eventMetadata, preMultipliedRotation)
+    };
+  }
+
+  private parseTransform(transformationData: Event360TransformationData, preMultipliedRotation: boolean): Matrix4 {
+    const translationComponents = transformationData.translation.split(',').map(parseFloat);
+    if (translationComponents.length !== 3 || translationComponents.some(isNaN)) {
+      throw new Error(
+        `Invalid translation data: expected 3 numeric components, got "${transformationData.translation}"`
+      );
+    }
+
+    const rotationAxisComponents = transformationData.rotation_axis.split(',').map(parseFloat);
+    if (rotationAxisComponents.length !== 3 || rotationAxisComponents.some(isNaN)) {
+      throw new Error(
+        `Invalid rotation_axis data: expected 3 numeric components, got "${transformationData.rotation_axis}"`
+      );
+    }
+
+    const rotationAngle = parseFloat(transformationData.rotation_angle);
+    if (isNaN(rotationAngle)) {
+      throw new Error(
+        `Invalid rotation_angle data: expected numeric value, got "${transformationData.rotation_angle}"`
+      );
+    }
+
+    const millimetersInMeters = 1000;
+    const translation = new Vector3(
+      translationComponents[0],
+      translationComponents[2],
+      -translationComponents[1]
+    ).divideScalar(millimetersInMeters);
+
+    const rotationAxis = new Vector3(rotationAxisComponents[0], rotationAxisComponents[2], -rotationAxisComponents[1]);
+
+    const rotationAngleRadians = MathUtils.DEG2RAD * rotationAngle;
+    const rotationMatrix = new Matrix4().makeRotationAxis(rotationAxis, rotationAngleRadians);
+
+    const translationMatrix = new Matrix4().makeTranslation(translation.x, translation.y, translation.z);
+
+    const entityTransform = translationMatrix.clone();
+    if (!preMultipliedRotation) {
+      entityTransform.multiply(rotationMatrix.clone().multiply(new Matrix4().makeRotationY(Math.PI / 2)));
+    } else {
+      entityTransform.multiply(new Matrix4().makeRotationY(Math.PI));
+    }
+
+    return entityTransform;
+  }
+}
+function isEvent360Metadata(metadata: Metadata | undefined): metadata is Event360Metadata {
+  return (
+    metadata !== undefined &&
+    typeof metadata === 'object' &&
+    typeof metadata.site_id === 'string' &&
+    typeof metadata.station_id === 'string' &&
+    typeof metadata.rotation_angle === 'string' &&
+    typeof metadata.rotation_axis === 'string' &&
+    typeof metadata.translation === 'string'
```

Comment:
[Optional]: I would personally prefer just doing `in`-checks here (e.g. `'site_id' in metadata`), since the `Metadata` type already asserts that all the fields are of type string. But this is fine if you prefer it